src/main/scala/io/iohk/ethereum/jsonrpc/DebugService.scala

```diff
@@ -25,9 +24,9 @@ class DebugService(peerManager: ActorRef, etcPeerManager: ActorRef) {
     val result = for {
       ids <- getPeerIds
       peers <- Future.traverse(ids)(getPeerInfo)
-    } yield ListPeersInfoResponse(peers.flatten)
+    } yield Right(ListPeersInfoResponse(peers.flatten))
 
-    result.map(Right(_))
+    Task.fromFuture(result)
```

Comment:
`Task.fromFuture` takes a by-name parameter to ensure ref transparency. Having a variable holding `Future` breaks it and may lead to weird error where something is already executed despite missing `run` on `Task`

Reply:
Good point. This is fixed now.